File: params.f90
module params
    use, intrinsic :: iso_fortran_env

    implicit none
    
    integer, parameter :: dp = real64
    integer, parameter :: sp = real32

    ! Frequency with which to write output
    integer, parameter :: write_freq = 50

    ! Output filename
    character(len=20) :: outfile = 'output.nc'

    ! Write values of X variables in output file?
    logical, parameter :: output_x = .true.

    ! Write values of Y variables in output file?
    logical, parameter :: output_y = .false.

    !Start and end times, and number of time steps
    real(dp), parameter :: dt = 0.005_dp
    real(dp), parameter :: fin = 12_dp
    integer, parameter :: n_steps = fin / dt

    ! State vector properties
    integer, parameter :: n_x = 8
    integer, parameter :: n_y = 8
    integer, parameter :: n_z = 8
    
    ! State vector dimension of full model
    integer, parameter :: truth_dim = n_x + n_x*n_y + n_x*n_y*n_z
    
    ! State vector dimension of parametrised model
    integer, parameter :: state_dim = n_x + n_x*n_y
    
    ! Number of ensemble members
    integer, parameter :: n_ens = 40

    ! Frequency of assimilations, i.e. 1 = every timestep
    integer, parameter :: assim_freq = 1

    ! Observation error variance
    real(dp), parameter :: y_var = 0.1_dp

    ! 1 = observe all Y variables, 2 = observe every other, 4 = observe every 4th
    integer, parameter :: y_skip = 1

    ! Dimension of observation vector
    integer, parameter :: obs_dim = n_x*n_y/y_skip

    ! Default significand
    integer, parameter :: sbits = 10

    ! Covariance inflation factor
    real(dp), parameter :: rho = 1.02_dp

    ! Covariance localisation length scales
    real(dp), parameter :: loc = 1.0_dp
    real(dp), parameter :: loc_y = loc*26
    real(dp), parameter :: loc_x = loc*8
end module params

File: params.jl
# Frequency with which to write output
const write_freq = 1

# Output filename
const outfile = "output.nc"

# Time step
const Δt = 0.005

# Start and end times, and number of time steps
const fin = 2.0
const nsteps = Int(fin / Δt)

# State vector properties
const nx = 8
const ny = 8
const nz = 8

# State dimension of full model
const truthdim = nx + nx*ny + nx*ny*nz

# State dimension of parametrised model
const statedim = nx + nx*ny

# Number of ensemble members
const nens = 400

# Frequency of assimilations, i.e. 1 = every timestep
const assim_freq = 10

# Observation error variance
const σ² = 0.1

# Dimension of observation vector
const obsdim = nx*ny

# Covariance inflation factor
const ρ = 1.15